jetty-core/jetty-security/src/main/java/org/eclipse/jetty/security/MultiAuthenticator.java

```diff
@@ -0,0 +1,457 @@
+package org.eclipse.jetty.security;
+
+import java.io.Serial;
+import java.io.Serializable;
+import java.security.Principal;
+import java.util.function.Function;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.http.pathmap.MatchedResource;
+import org.eclipse.jetty.http.pathmap.PathMappings;
+import org.eclipse.jetty.http.pathmap.PathSpec;
+import org.eclipse.jetty.security.authentication.LoginAuthenticator;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.server.Session;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.URIUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>An {@link Authenticator} which maps different {@link Authenticator}s to {@link PathSpec}s.</p>
+ * <p>This can be used to support multiple different authentication methods for a single application such as
+ * FORM, OPENID and SIWE.</p>
+ * <p>The {@link #setLoginPath(String)} can be used to set a login page where unauthenticated users are
+ * redirected in the case that no {@link Authenticator}s were matched. This can be used as a page to
+ * link to other paths where {@link Authenticator}s are mapped to so that users can choose their login method.</p>
+ */
+public class MultiAuthenticator extends LoginAuthenticator
+{
+    private static final Logger LOG = LoggerFactory.getLogger(MultiAuthenticator.class);
+    public static final String LOGIN_PATH_PARAM = "org.eclipse.jetty.security.multi.login_path";
```

Comment:
Make it `private`.

Also, group together `static` fields, then empty line, then non-`static` fields.